codec/reflectcodec/type_codec.go

```diff
@@ -85,12 +91,17 @@ func (c *genericCodec) Size(value interface{}) (int, error) {
 		return 0, errMarshalNil // can't marshal nil
 	}
 
-	size, _, err := c.size(reflect.ValueOf(value))
+	metaValue := reflect.ValueOf(value)
+	if metaValue.Kind() == reflect.Ptr {
+		metaValue = metaValue.Elem()
+	}
```

Comment:
What is this change doing? And is it required?

Reply:
This is a tiny optimization, because the value is first a pointer instead of doing a call to dereference the pointer I did that in-place. That was needed _before_ but now that `omitempty` is required this bit can be removed, but this is a nice optimization to avoid a recursion call. Your call @StephenButtolph. I can safely remove it now.

Reply:
I think we can move this out into a separate PR if it isn't needed here.... If we really wanted to - we could re-write this codec to only recurse for non-tail recursive calls. Which I think would be a much more general approach for this optimization.